File: scrap/priceCompare.js
// Importer la bibliothèque Axios pour effectuer des requêtes HTTP
const axios = require('axios');
// Importer la bibliothèque Cheerio pour analyser le HTML
const cheerio = require('cheerio');

// Fonction pour effectuer une recherche sur Leroy Merlin
async function searchOnLeroyMerlin(product) {
  const leroyMerlinURL = `https://www.leroymerlin.fr/search?q=${product}`;
  try {
    // Effectuer une requête HTTP GET
    const response = await axios.get(leroyMerlinURL);
    if (response.status === 200) {
      // Analyser le HTML de la page
      const $ = cheerio.load(response.data);
      const titles = $('.km-tile-designation');
      const subtitles = $('.mu-ml-050');
      const prices = $('.js-main-price');

      // Afficher les résultats
      console.log('Résultats de la recherche du concurrent 1:');
      titles.each((index, element) => {
        const title = $(element).text();
        const subtitle = $(subtitles[index]).text();
        const price = $(prices[index]).text();
        console.log(title);
        console.log(subtitle);
        console.log(price);
        console.log('__________');
      });
    } else {
      console.log('Erreur lors de la requête HTTP.');
    }
  } catch (error) {
    console.error('Une erreur s\'est produite :', error);
  }
}

// Obtenir l'entrée de l'utilisateur
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Entrez le produit recherché: ', (product) => {
  searchOnLeroyMerlin(product);
  rl.close();
});

File: scrap/priceCompare.py
"""    Learn python 
      github/berru-g 23
      Outil pour comparer les prix de vente de ses concurents
"""
#next step: interface web
import time
import requests
from bs4 import BeautifulSoup

print("Entrez le produit recherché:")
product = input("")

first_url = "https://www.leroymerlin.fr/search?q={}"

# Search first
first_search = requests.get(first_url.format(product))
print("request")
first_soup = BeautifulSoup(first_search.text, "html.parser")
print("parser")
first_titles = first_soup.find_all(class_="km-tile-designation")
print("title")
subtitles = first_soup.find_all(class_="mu-ml-050")
print("sub")
prices = first_soup.find_all(class_="js-main-price")
print("price")

print("Résultats de la recherche du concurent 1:")
 
# Print the titles
for title in first_titles:
    for subtitle in subtitles:
        for price in prices:
            print(title.text)
            print(subtitle.text) 
            print(price.text) 
            print("__________")
time.sleep(2)
 
print("Fin des résultats")
print("Si aucun resultat n'est trouvé alors changer le nom des class dans le script")
print("_______________")
time.sleep(4)
exit()
